pkg/command/buildbase.go

```diff
@@ -71,6 +88,21 @@ func (opts *BaseBuildOptions) Validate(cmd *cobra.Command, args []string) error
 	return nil
 }
 
-func (opts *BaseBuildOptions) tag() (name.Tag, error) {
-	return name.NewTag(opts.ImageName, name.WeakValidation)
+type imageNameContext struct {
+	url.URL
```

Comment:
Do you anticipate ever including more stuff in here? Embedding this seems fine, especially for brevity, but I can imagine name collisions if you want to include more. I guess you can namespace any additions under their own field name, though, so it doesn't _really_ matter.

Reply:
I wanted to leave the door open, but am not 100% sure if/what that might be.